contracts/proposers/BurnAuction.sol

```diff
@@ -0,0 +1,101 @@
+pragma solidity ^0.5.15;
+import { Chooser } from "./Chooser.sol";
+import { Bitmap } from "../libs/Bitmap.sol";
+import { Ownable } from "@openzeppelin/contracts/ownership/Ownable.sol";
+
+contract BurnAuction is Chooser, Ownable {
```

Comment:
Diff to iden3: We let a proposer call rollup contract, which then calls burn auction contract. Iden3 let proposer call burn auction first then rollup contract next.